query/src/main/scala/filodb/query/exec/SelectRawPartitionsExec.scala

```diff
@@ -32,64 +32,67 @@ final case class SelectRawPartitionsExec(id: String,
                                          submitTime: Long,
                                          limit: Int,
                                          dispatcher: PlanDispatcher,
-                                         dataset: DatasetRef,
+                                         datasetRef: DatasetRef,
                                          shard: Int,
+                                         dataSchema: Schema,
                                          filters: Seq[ColumnFilter],
                                          chunkMethod: ChunkScanMethod,
                                          colIds: Seq[Types.ColumnId]) extends LeafExecPlan {
   import SelectRawPartitionsExec._
 
-  protected[filodb] def schemaOfDoExecute(dataset: Dataset): ResultSchema = {
+  def dataset: DatasetRef = datasetRef
+
+  protected[filodb] def schemaOfDoExecute(): ResultSchema = {
     require(!colIds.contains(0),
       "User selected columns should not include timestamp (row-key); it will be auto-prepended")
 
-    val selectedColIds = selectColIds(dataset)
+    val selectedColIds = selectColIds()
```

Comment:
Note that this will only return colIds required for serving the downsampled data. The inference of resultSchema == dsGuage later in the pipeline would be wrong.

Reply:
Yes, this will be fixed in subsequent PR.